packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -1012,6 +1012,18 @@ class _DatePickerHeader extends StatelessWidget {
   }
 }
 
+/// Signature for predicating enabled dates in date range pickers.
+///
+/// The [selectedStartDay] and [selectedEndDay] are the currently selected start
+/// and end dates of a date range, which conditionally enables or disables each
+/// date in the picker based on the user selection. (Example: in a hostel's room
+/// selection, you are not able to select the end date after the next
+/// non-selectable day).
+///
+/// See [showDateRangePicker], which has a [SelectableDayForRangePredicate]
+/// parameter used to specify allowable days in the date range picker.
+typedef SelectableDayForRangePredicate = bool Function(DateTime day, DateTime? selectedStartDay, DateTime? selectedEndDay);
```

Comment:
What do you think about (in a separate PR) adding an example on a use case for this parameter? It feels complex enough to warrant one.

Reply:
Where should I put the example, in the comment and/or somewhere else?

Reply:
I usually see examples like this in the docs for the widget, but all these conversations are probably for a (possible) subsequent PR.